test/go/table_schema_cache_test.go

```diff
@@ -65,30 +76,159 @@ func (this *TableSchemaCacheTestSuite) TestLoadTablesWithoutFiltering() {
 }
 
 func (this *TableSchemaCacheTestSuite) TestLoadTablesRejectTablesWithoutNumericPK() {
-	query := fmt.Sprintf("CREATE TABLE %s.%s (id varchar(20) not null, data TEXT, primary key(id))", testhelpers.TestSchemaName, "test_table_4")
+	table := "test_table_4"
+	paginationColumn := "id"
+	query := fmt.Sprintf("CREATE TABLE %s.%s (%s varchar(20) not null, data TEXT, primary key(%s))", testhelpers.TestSchemaName, table, paginationColumn, paginationColumn)
+	_, err := this.Ferry.SourceDB.Exec(query)
+	this.Require().Nil(err)
+
+	_, err = ghostferry.LoadTables(this.Ferry.SourceDB, this.tableFilter, nil, nil, nil)
+
+	this.Require().NotNil(err)
+	this.Require().EqualError(err, ghostferry.NonNumericPaginationKeyError(testhelpers.TestSchemaName, table, paginationColumn).Error())
+	this.Require().Contains(err.Error(), table)
+}
+
+func (this *TableSchemaCacheTestSuite) TestLoadTablesCascadingPaginationColumnConfigRightScenario1() {
+	table := "pagination_by_column_config_right_scenario_1"
+	paginationColumn := "identity"
+	cascadingPaginationColumnConfig := &ghostferry.CascadingPaginationColumnConfig{
+		PerTable: map[string]map[string]string{
+			testhelpers.TestSchemaName: map[string]string{
+				table: paginationColumn,
+			},
+		},
+	}
+
+	query := fmt.Sprintf("CREATE TABLE %s.%s (%s bigint(20) not null, data TEXT)", testhelpers.TestSchemaName, table, paginationColumn)
+	_, err := this.Ferry.SourceDB.Exec(query)
+	this.Require().Nil(err)
+
+	_, err = ghostferry.LoadTables(this.Ferry.SourceDB, this.tableFilter, nil, nil, cascadingPaginationColumnConfig)
+
+	this.Require().Nil(err)
+}
+func (this *TableSchemaCacheTestSuite) TestLoadTablesCascadingPaginationColumnConfigRightScenario2() {
+	dropTestTables(this) // needed because they interfere with this test
```

Comment:
Can you elaborate what "they" means? It's unclear to me. Shouldn't this method be in the test setup method instead?

Reply:
The test setup introduces different tables: `test_table_1, _2, _3, and _4` Those tables are interfering with this test

Reply:
I'm a bit confused why the following code doesn't already address this problem as it drops the entire schema.

https://github.com/Shopify/ghostferry/blob/b8147c5ff8653099f6c5e24128aa6fd41ff401e1/testhelpers/unit_test_suite.go#L81-L90

Reply:
I only need to drop the tables. The database can remain. 
If you feel that there is merit in this helper function, would you suggest me to promote it to `unit_test_suite.go`?

Reply:
No I meant the code I linked should have ran before your test so you shouldn't need to drop additional tables since you are using the same schema. If that is the case you shouldn't have needed this function at all.

Just leave the code here for now if you dont want to dig. We can make a note to deal with it later.

Reply:
I am not following yet. The TestSetup does: https://github.com/Shopify/ghostferry/blob/97d62446c2f7d978cb6e727e3024bdddd27f4e3f/test/go/table_schema_cache_test.go#L42-L45
which always is called by: https://github.com/Shopify/ghostferry/blob/97d62446c2f7d978cb6e727e3024bdddd27f4e3f/test/go/table_schema_cache_test.go#L338-L341 before each test. 

So by the time the test: `TestLoadTablesCascadingPaginationColumnConfigRightScenario2()` is invoked, those 4 tables are present and populated.

I cannot run `dropTestTables` inside the setup because other tests rely on the existence of those tables, e.g. `test_table_1, _2, _3, and _4`. 

What am I misunderstanding?